packages/theme/modules/catalog/category/components/views/CategoryProductPrice.vue

```diff
@@ -0,0 +1,95 @@
+<template>
+  <div
+    v-if="product.__typename === 'GroupedProduct'"
+    class="product-price"
+  >
+    <div class="product-price__row">
+      <span
+        class="product-price__label"
+      >{{ $t('Starting at') }}:</span>
+      <SfPrice
+        class="sf-product-card__price"
+        :regular="product.commonProps.regularPrice"
+        :special="product.commonProps.specialPrice"
+      />
+    </div>
+  </div>
+  <div
+    v-else-if="product.__typename === 'BundleProduct'"
+    class="product-price"
+  >
+    <div class="product-price__row">
+      <span
+        class="product-price__label"
+      >{{ $t('From') }}:</span>
+      <SfPrice
+        class="sf-product-card__price"
+        :regular="product.commonProps.regularPrice"
+        :special="product.commonProps.specialPrice"
+      />
+    </div>
+    <div class="product-price__row">
+      <span
+        class="product-price__label"
+      >{{ $t('To') }}:</span>
+      <SfPrice
+        class="sf-product-card__price"
+        :regular="product.commonProps.maximumPrice"
+      />
+    </div>
+  </div>
+  <div
+    v-else
+    class="product-price"
+  >
+    <SfPrice
+      class="sf-product-card__price"
+      :regular="product.commonProps.regularPrice"
+      :special="product.commonProps.specialPrice"
+    />
+  </div>
+</template>
+<script lang="ts">
+import { defineComponent, PropType } from '@nuxtjs/composition-api';
+import { SfPrice } from '@storefront-ui/vue';
+import {
+  useProductsWithCommonProductCardProps,
+} from '~/modules/catalog/category/components/views/useProductsWithCommonCardProps';
+
+export default defineComponent({
+  name: 'CategoryProductPrice',
+  components: {
+    SfPrice,
+  },
+  props: {
+    product: {
+      type: Object as PropType<typeof useProductsWithCommonProductCardProps | null>,
```

Comment:
```suggestion
            type: Object as PropType<ReturnType<typeof useProductsWithCommonProductCardProps>['productsWithCommonProductCardProps']['value'][number] | null>,
```
Unfortunately this is the type you're looking for. Right now in this PR it's "the product prop is a function or null (typeof useProduct... is a function that accepts products)

What I suggested is the modified product returned by the commoncardprops

Reply:
Awful...